pydm/main_window.py

```diff
@@ -111,6 +111,8 @@ def join_to_current_file_path(self, ui_file):
 
     def open_file(self, ui_file, macros=None, command_line_args=None):
         filename = self.join_to_current_file_path(ui_file)
+        if not os.path.exists(filename):
+            raise IOError("File {} not found".format(filename))
```

Comment:
I'm against any method we will call from within `Qt` to raise an Exception. We don't ever want to bring everything crashing down on the user. Can we just report an `logger.error` here?

I realize that this doesn't let `PyDMRelatedButton` show the error message in the window, but hopefully someday we will pipe the `logging` messages into the `StatusBar` globally.

Reply:
Well... we can totally move the error messaging from the `PyDMRelatedButton` to here... 
Which would include the `logging` and `StatusBar`... what do you think about that?

Reply:
Instead of passing the status bar, how about returning a status message? An empty message means no error (or you can provide a default content, something like "No error." You'll display this message no matter what on the window.